### PR TITLE
[WIP] Refactor/chat views and subs

### LINKED ISSUE #6033: Remove logic from desktop messages list view
### User Story

As a developer, I want to remove logic from view

### Description

*Type*: Improvement

*Summary*: Discussed with @yenda that having logic inside view in `src/status_im/ui/screens/desktop/main/chat/views.cljs` isn't ok. At the moment of creating PR #6006 it wasn't clear how to remove it, so this issue created for later fix.

#### Expected behavior
View doesn't contain logic

#### Actual behavior
View contains logic

### Reproduction
- Open `src/status_im/ui/screens/desktop/main/chat/views.cljs`
- check `messages-view` implementation

### Solution
[comment]: # (Please summarise the solution and provide a task list on what needs to be fixed.)
*Summary*:  TBD

### Additional Information
[comment]: # (Please do your best to fill this out.)
* Status version: V0.0.1 (ed2abf910)
[comment]: # (Android or iOS.)
* Operating System: Mac

### PR DESCRIPTION
**Pull request by yenda**
[WIP] cutting the PR in smaller pieces

PR is long because:
- most of chat.subs and contact.subs code is moved to chat.db and contact.db
- utils.contacts is moved to contact.db
- lots of logic is moved from views to subs
- tests are touched

replaces #6652 partially (block feature will be added separately to reduce size of PR)
fix #6033 remove logic from view

- [fix] usernames in bold font on Linux
- [fix] current chat in bold font Linux
- [refactor] remove defview and subscriptions to
use pure functions with params, use flatlist on
desktop for chats like it is done in mobile
- [refactor] optimize subscriptions for perf, remove
message-groups
- [fix] prevent use of rtl char in username to impersonate
other other user random name

### Summary:

Simplify views and subscriptions to facilitate implementation of block contact feature and other future improvements.

Only root views are subscribing and using `defview` macro, children use reagent form-1 components as much as possible.
Priority map is used to keep messages of chats in order without having to sort them every time so that only a single reduce pass is needed to enrich them with contextual data.

#### Platforms
- Android
- iOS
- macOS
- Linux
- Windows

#### Areas that maybe impacted
**Functional**
- 1-1 chats
- public chats
- group chats

**Non-functional**
- battery performance
- CPU performance / speed of the app
- network consumption

status: wip (still have to check if mobile works)

### CODE REVIEW

**Comment by rasom** [comment]
@yenda is it possible to implement #6652 without moving code from/to different namespaces? Has it been discussed?

File: src/status_im/utils/priority_map.cljs [hunk 1]
```diff
@@ -0,0 +1,229 @@
+(ns status-im.utils.priority-map
```
Comment (rasom) [inline src/status_im/utils/priority_map.cljs]: @yenda is it copied from somewhere or that's yours implementation?

Comment (yenda) [reply src/status_im/utils/priority_map.cljs]: yes it's https://github.com/tailrecursion/cljs-priority-map/blob/master/src/cljs/tailrecursion/priority_map.cljs with a bug fixv for keys and val fns

**Review by dmitryn — APPROVED** [review]

File: src/status_im/chat/models/loading.cljs [hunk 2]
```diff
@@ -70,17 +39,17 @@
...
+                                                                        (get-referenced-messages
+                                                                         chat-id
+                                                                         (get-referenced-ids chat-messages))))))))
                       {}
                       all-stored-chats)]
     (fx/merge cofx
               {:db (assoc db
                           :chats          chats
                           :contacts/dapps default-dapps)}
-              (group-messages)
```
Comment (rcullito) [inline src/status_im/chat/models/loading.cljs:84]: I know you mention it in your PR summary, but why are these getting removed? group messages are no longer being supported across the app?

File: src/status_im/constants.cljs [hunk 3]
```diff
@@ -18,7 +18,7 @@
 (def min-password-length 6)
 (def max-chat-name-length 20)
 (def response-suggesstion-resize-duration 100)
-(def default-number-of-messages 20)
+(def default-number-of-messages 100)
```
Comment (rcullito) [inline src/status_im/constants.cljs:21]: this will be # of messages loaded initially?

File: src/status_im/ui/screens/chat/message/message.cljs [hunk 4]
```diff
@@ -41,22 +37,22 @@
...
-  (letsubs [username [:get-contact-name-by-identity from]]
-    [react/view {:style (style/quoted-message-container outgoing)}
-     [react/view {:style style/quoted-message-author-container}
-      [vector-icons/icon :icons/reply {:color (if outgoing colors/wild-blue-yonder colors/gray)}]
-      [react/text {:style (style/quoted-message-author outgoing)}
-       (chat.utils/format-reply-author from username current-public-key)]]
-     [react/text {:style           (style/quoted-message-text outgoing)
-                  :number-of-lines 5}
-      text]]))
+(defview quoted-message
```
Comment (rcullito) [inline src/status_im/ui/screens/chat/message/message.cljs]: can this become a pure fn or does it need to stay as a defview?

File: src/status_im/ui/screens/chat/views.cljs [hunk 5]
```diff
@@ -1,143 +1,139 @@
...
+      (toolbar/nav-back-count {:home? true}))
+    [toolbar-content/toolbar-content-view current-chat]
+    (when-not modal?
+      [toolbar/actions [{:icon      :icons/options
+                         :icon-opts {:color               :black
+                                     :accessibility-label :chat-menu-button}
+                         :handler   #(on-options chat-id name group-chat public?)}]])]
+   (when-not (or public? group-chat) [add-contact-bar contact])])
 
-(defmulti message-row (fn [{{:keys [type]} :row}] type))
```
Comment (rcullito) [inline src/status_im/ui/screens/chat/views.cljs:68]: any conceptual reason behind this change from *message-row* to *message-view*?

File: src/status_im/ui/screens/desktop/main/chat/styles.cljs [hunk 6]
```diff
@@ -38,6 +38,7 @@
 
 (def author
   {:color         colors/black
+   :font-weight   :bold
```
Comment (rcullito) [inline src/status_im/ui/screens/desktop/main/chat/styles.cljs:41]: :tada: :balloon:

**Review by rcullito — APPROVED** [review]
big overhaul! your changes definitely improved readability for someone not super familiar with the codebase. a couple of questions inline on concepts and various naming conventions.  :+1:

**Comment by oskarth** [comment]
I find it difficult to believe that such a big PR won't have a bunch of unforseen regressions. Can we make this diff smaller and just address minimal necessary to solve the problem at hand, instead of cramming a bunch of stuff together?

`git add -p` is your friend!

File: src/status_im/constants.cljs (hunk 3 above)
Comment (yenda) [reply src/status_im/constants.cljs:21]: per chat, but good catch I'll try less 100 might be too much it was to smooth scrolling

File: src/status_im/ui/screens/chat/views.cljs (hunk 5 above)
Comment (yenda) [reply src/status_im/ui/screens/chat/views.cljs:68]: just to align with mobile

File: src/status_im/chat/models/loading.cljs (hunk 2 above)
Comment (yenda) [reply src/status_im/chat/models/loading.cljs:84]: this is not group chat, it was grouping of messages by datemark, done at message reception for caching, but now the subscriptions are more efficient and messages need to be filtered alter for tag features so the cache will too often be invalid to be useful

File: src/status_im/ui/screens/chat/message/message.cljs (hunk 4 above)
Comment (yenda) [reply src/status_im/ui/screens/chat/message/message.cljs]: should be a defn thanks

File: src/status_im/utils/priority_map.cljs (hunk 1 above)
Comment (yenda) [reply src/status_im/utils/priority_map.cljs]: do you think I should clone the repo instead @rasom ?

**Comment by lukaszfryc** [comment]
@churik could you help me with testing this PR? Especially on Linux and Windows. I can take mobile + macOS. 

> [fix] usernames in bold font on Linux
[fix] current chat in bold font Linux

**Comment by churik** [comment]
@lukaszfryc I'm not sure it is ready according to https://github.com/status-im/status-react/pull/6764#issuecomment-439765593 and Roman's comments.
WDYT?

**Comment by lukaszfryc** [comment]
@yenda, should we wait with testing?

**Comment by yenda** [comment]
@lukaszfryc yes please wait

File: src/status_im/chat/models/loading.cljs (hunk 2 above)
Comment (rcullito) [reply src/status_im/chat/models/loading.cljs:84]: thanks for clarifying.

File: src/status_im/ui/screens/chat/views.cljs (hunk 5 above)
Comment (rcullito) [reply src/status_im/ui/screens/chat/views.cljs:68]: got it. thanks.

**Comment by lukaszfryc** [comment]
@yenda, any update?

**Comment by yenda** [comment]
@lukaszfryc I will close this PR once I'm done moving all the changes to separate PRs

**Pull request closed without merging**